### DISCUSSION TITLE
[CDK] add parent stream field to child URL path

### DISCUSSION THREAD

**Discussion by pasunboneleve** [airbytehq/airbyte#33739] [Connector Ideas and Features]
1. I have a source endpoint `/{{ config['clientnamespace'] }}/v1/Employees` that returns an array of `employee_id`;
2. I have another source endpoint `/{{ config['clientnamespace'] }}/v1/Employees/{{ employee_id }}/Schedules` that returns employee schedules.

I need to fetch all employee schedules.

This is a common scenario. I'm developing a connector for this external API, and Airbyte lets me inject fields from the parent stream into the child stream, but only as query parameters, header or body of the request. Not in the URL itself.

I believe this is desirable functionality. Is there a way to make this work with the current implementation, or were I should I go from here? I created an issue for this [here](https://github.com/airbytehq/airbyte/issues/33738).

**Comment by marcosmarxm** [comment]
Are you using the connector builder or the low-code CDK?

**Comment by pasunboneleve** [comment]
I’m using the web ui. Is that the connector builder? It is referred to by the docs as no-code builder.

**Reply by rshorser** [reply]
+1 to this, I'm running into the same constraint in the no-code connector builder on the UI. Based on the tooltip in the builder, I tried using the `stream_partition` inside the URL path but it doesn't seem to be getting populated when I test it in the builder itself. 

E.g, in the url of the child stream, I put a URL that looks like this
`https://somehost.com/api/v1/employees/{{stream_partition['id']}}/details` but when I press "Test connection", it just sends the request to a URL like this `https://somehost.com/api/v1/employees//details`

**Reply by pasunboneleve** [reply]
Looks like the correct syntax is `{{ stream_partition.id }}`. The docs we looked at seem to have been written with **Python** in mind, while the UI works with **JavaScript**. 😅

**Comment by pasunboneleve** [comment]
Oh, I found it hidden in the [docs](https://docs.airbyte.com/connector-development/connector-builder-ui/partitioning/#parent-stream). Trying to make it work now.

[a comment by an automated account is not shown]
